walker.go

```diff
@@ -268,6 +277,14 @@ func Walk(ctx context.Context, p string, opt *WalkOpt, fn filepath.WalkFunc) err
 
 		dir.calledFn = true
 
+		fi, err := dirEntry.Info()
```

Comment:
Can this be a duplicate call for the same file that already had lstat call in 165?

Looks like the value is not cached and always results in a syscall. https://github.com/golang/go/blob/master/src/os/file_unix.go#L472

Reply:
ya, good call. it makes a slight difference.

Reply:
Are you planning to make that change? Also not the skipdir handler in this error return.